FHEM/14_SD_WS.pm

```diff
@@ -125,6 +127,49 @@ sub SD_WS_Undef {
   my ($hash, $name) = @_;
   carp "SD_WS_Undef, too few arguments ($hash, $name)" if @_ < 2;
   delete($modules{SD_WS}{defptr}{$hash->{CODE}}) if(defined($hash->{CODE}) && defined($modules{SD_WS}{defptr}{$hash->{CODE}}));
+	RemoveInternalTimer($hash);
+  return;
+}
+
+#############################
+sub SD_WS_Set {
+  my ($hash, $name, $cmd, @args) = @_;
+  my $ret = undef;
+	my $LastInputDev = $hash->{LASTInputDev};
+  return $ret if (!defined $LastInputDev);
+	
+  my $rhash = $defs{$LastInputDev};
+  my $rlongids = AttrVal($rhash->{NAME},'longids',0);
+	my $def = $hash->{DEF};
+	my @array=split(/_/,$def);
```

Comment:
Vorschlag, splitten mit Limit auf drei Elemente und anschließend join:
```suggestion
	my $modes = join ('_', split(/_/,$def,3));
```

Reply:
Das funktioniert so nicht. Im letzten Element des Arrays befindet sich dann immer der Rest des Strings:

```
2022.02.06 13:23:50 3: SD_WS_33_T_5: SD_WS_Set, 1 model=SD_WS_33
2022.02.06 13:23:50 3: SD_WS_33_T_5: SD_WS_Set, 2 model=SD_WS_33_T_18_2
```

Zeile 1 ist die Variante von mir.
Zeile 2 ist deine.

Siehe auch: https://perldoc.perl.org/functions/split
```
my @x = split(//, "abc", 1); # ("abc")
my @x = split(//, "abc", 2); # ("a", "bc")
```

Reply:
:(

Reply:
also join ('_',$array[0],$array[1],$array[2]) um nur drei Werte zu verbinden.

Reply:
oder  mit einer Regex :) Lass es halt so wie Du es hattest. Ich dachte nur es ist seltsam, da alles in einen array gepackt wird und dann wieder zusammen gesetzt wird.
`(\w\w_\w\w_\w\w)/ `